src/main/java/net/dv8tion/jda/internal/requests/restaction/pagination/ThreadChannelPaginationActionImpl.java

```diff
@@ -46,23 +50,27 @@ public EnumSet<PaginationOrder> getSupportedOrders()
         return EnumSet.of(PaginationOrder.BACKWARD);
     }
 
+    //Thread pagination supplies ISO8601 timestamps for some cases, see constructor
+    @Nonnull
     @Override
-    protected Route.CompiledRoute finalizeRoute()
+    protected String getPaginationLastEvaluatedKey(long lastId, ThreadChannel last)
     {
-        Route.CompiledRoute route = super.finalizeRoute();
-
-        final String limit = String.valueOf(this.limit.get());
-        final long last = this.lastKey;
+        if (useID)
+            return Long.toUnsignedString(lastId);
 
-        route = route.withQueryParams("limit", limit);
+        if (order == PaginationOrder.FORWARD && lastId == 0)
+        {
+            // first second of 2015 aka discords epoch, hard coding something older makes no sense to me
+            return "2015-01-01T00:00:00.000";
+        }
 
-        if (last == 0)
-            return route;
+        // this should be redundant, due to calling this with PaginationAction#getLast() as last param,
+        // but let's have this here.
+        if (last == null)
+            throw new NoSuchElementException("No entities have been retrieved yet.");
```

Comment:
This could be replaced with the current ISO8601 timestamp instead of throwing an exception